Many-updates-in-one pull request...

Gyosa3 opened the pull request:
Hi, when the stop queried is the first/last one of the line, there are chances that there are duplicated records, for the tram/bus that arrives and for when it leaves. I have seen this for example for this code:

https://v0.ovapi.nl/tpc/31008732

The same metro arrives and leaves the other way with the same route code and the same time, therefore I propose to ignore times of departure when the stop is the last stop of the line, i.e. the 'JourneyStopType' attribute is 'LAST'.

For me this cleanly deduplicates the timings at this stop. I tested the change locally and it works fine for me.

Gyosa3 committed:
Update sensor.py

Filter out times of arrival at final destination to prevent double entries at end-of-line stops

Gyosa3 committed:
Update sensor.py

Gyosa3 committed:
Update README.md

added details for the install steps

Gyosa3 committed:
Add uniqueID to entities

Enable the entities to be updated and deleted in the UI

Gyosa3 committed:
cosmetic improvements

Fixed some details on displayed attributes

Gyosa3 committed:
Fix display details

remove stop code, route code and timing code to have a single attribute that shows only the relevant codes

Gyosa3 commented:
I have added all my fixes in the same pull request. It works on my setup, but please test on your side!

I have noticed that you were not showing the delay on the "master" entity, i.e. the first sensor. I reverted this to have the same display on all entities, but it's just a personal taste.

For example the first instance here shows delay as well as the "future" sensors. Also it now shows negative delay when the tram arrives early (yes, sometimes!).
![image](https://github.com/Paul-dH/Home-Assisant-Sensor-OvApi/assets/51777889/0aa322da-1aad-47a5-b40b-55359d3c477d)

Same with real delay, the first instance now shows the delay instead of showing the actual arrival time. the reason why I reverted for this is that by keeping the original arrival time you can still recognise if it is the tram or bus that you usually expect, not the previous one that was very late (like "19:30" could be the tram from "19:25 + 5m" or the tram from "19:32 -2m" so you would better know in which tram you get in).

I also created a "Search_code" attribute that either displays the Timing point code or the stop_code + route_code depending on the setup of the sensor:
![image](https://github.com/Paul-dH/Home-Assisant-Sensor-OvApi/assets/51777889/91f60ffa-3f41-410c-9091-1f75e17f2815)

or
![image](https://github.com/Paul-dH/Home-Assisant-Sensor-OvApi/assets/51777889/9fb0cf4c-d062-458b-b2e0-11c53596626d)

I also changed the display of the delay when there is no delay:
![image](https://github.com/Paul-dH/Home-Assisant-Sensor-OvApi/assets/51777889/7ec94849-1eb9-451f-9712-bd5f3ab2dfd7)

and when there is:
![image](https://github.com/Paul-dH/Home-Assisant-Sensor-OvApi/assets/51777889/0b4f0e19-65e0-4540-a8bd-59f1b1d7e560)

Paul-dH commented:
I've updated my lab environment and pulled HA out of the dust. I'm doing some upgrades now to bump to the latest version. I'll keep you posted on the progress...

Gyosa3 committed:
Fixed state for unknown schedules

Schedules after last transport will show '--:--' instead of 'unknown'

Gyosa3 commented:
great! I've added another small commit to this pull request to improve the state display for the 'unknown' schedules.

Gyosa3 committed:
Changed icon for tram

Gyosa3 commented:
That how it looks at the end of the day:
![image](https://github.com/Paul-dH/Home-Assisant-Sensor-OvApi/assets/51777889/0465d501-3f38-4488-8ba7-4852d91f2686)

Gyosa3 committed:
Update __init__.py

Gyosa3 committed:
Update __init__.py

Gyosa3 committed:
Update __init__.py

Gyosa3 committed:
Reload available in Dev tools screen

Gyosa3 committed:
Create services.yaml

Gyosa3 commented:
Hi again

I have implemented the reload service call in __init__.py and also I have created a new file services.yaml

I am not sure how this works, but at least it works and I can reload the sensors from the Developer tools page and it works without having to restart HA completely :)

So that's the end of my contribution everything works as I was expecting at the start. Hope you like it.
Cheers!

Gyosa3 committed:
Added count in minutes until transport arrives

This is a convenient attribute for remote displays to pre-calculate how many minutes remain until transport arrives. So you can start running...

Gyosa3 commented:
Ultimate change to add a new attribute that calculates how many minutes remain until transport arrives. Convenient to use in a remote display without having the remote chip to do the maths...

![image](https://github.com/Paul-dH/Home-Assisant-Sensor-OvApi/assets/51777889/393c9b94-0764-4493-ac91-b5ae95b7c57a)

Gyosa3 committed:
Update sensor.py

adjustment to the 'arrives' attribute to avoid showing 'in 0 min.' and increased component version after all

kirikorneev commented:
@Gyosa3 Thank you for this PR! There are unfortunately a few errors which were blocking the code to run:

- missing semicolon after `else` on lines 180 & 247
- a typo on line 293, `stop_list` instead of `stops_list`
